src/XUnity.AutoTranslator.Plugin.MelonMod/AutoTranslatorPlugin.cs

```diff
@@ -19,18 +19,58 @@
 
 namespace XUnity.AutoTranslator.Plugin.MelonMod
 {
-   public class AutoTranslatorPlugin : MelonLoader.MelonMod
+   public class AutoTranslatorPlugin : MelonLoader.MelonMod, IPluginEnvironment
    {
+      private ExIni.IniFile _file;
+      private string _configPath;
+      private string _dataFolder;
       public override void OnApplicationLateStart()
       {
 #if IL2CPP
          var modFi = new FileInfo( Location );
          var gameDir = modFi.Directory.Parent;
          var unhollowedPath = Path.Combine( gameDir.FullName, "MelonLoader", "Il2CppAssemblies" );
          Il2CppProxyAssemblies.Location = unhollowedPath;
+         _dataFolder = Path.Combine( gameDir.FullName, "AutoTranslator" );
+         _configPath = Path.Combine( _dataFolder, "Config.ini" );
+         TranslationPath =_dataFolder;
```

Comment:
Shouldn't this be done even outside of IL2CPP?

Reply:
True. My bad I included it inside the il2cpp block

Reply:
Now it doesn't compile, gameDir doesn't exist in non-il2cpp.

Reply:
Build tested.